### PR TITLE
refactor use cases html css js

### LINKED ISSUE #199: [REFACT] Rework Use Cases section for home page
To make it easier to support, let's reflect the use cases section on the home page: Image

### PR DESCRIPTION
**Pull request by AucT**
Closes #199 

PNYt2Uhc2X
uA1CcFYZDb

## Summary by CodeRabbit

- **New Features**
	- Introduced a new responsive CSS file for the use cases section, enhancing visual presentation and layout.
	- Added improved tabbed navigation functionality with keyboard accessibility for better user interaction with use cases.
	- Modularized the use cases content structure for better organization and maintainability.

- **Bug Fixes**
	- Removed outdated JavaScript files that were no longer needed, streamlining the application.

- **Chores**
	- Updated layout files to include new dynamic CSS and JavaScript resources, improving load performance.
	- Removed the inclusion of the jQuery library from the project.
	- Updated Ruby version in the Dockerfile for improved compatibility.

### CODE REVIEW

File: themes/beaver/assets/js/tabs.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+document.querySelectorAll('.js-tab').forEach(el => {
+    const container = el.closest('.js-tabs-container');
+    el.onclick = function (e) {
+        const panel = document.getElementById(e.target.getAttribute('aria-controls'));
+        container.querySelectorAll(".tab-panel").forEach(panel => panel.hidden = true);
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/js/tabs.js:6]: nope

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 2]
```diff
@@ -0,0 +1,150 @@
...
+.use-cases .tab-title {
+    background-color: #f5f6f8;
+    color: #121212;
+    border: none;
+    border-radius: 14px;
+    padding: 28px 35px;
+    font-size: 20px;
+    font-weight: 700;
+    cursor: pointer;
+    outline: none;
```
Comment (andriyParashchuk) [inline themes/beaver/assets/css/use-cases-dynamic.css]: I guess we should remove that Please see here https://gist.github.com/guilhermesimoes/4595862

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+}
+
+.d-flex {
+    display: flex !important;
+}
+.flex1 {
```
Comment (andriyParashchuk) [inline themes/beaver/assets/css/use-cases-dynamic.css]: class name is not clear. please check https://rstacruz.github.io/rscss/helpers.html
how to improve your selector naming

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+
+.use-cases .action-button {
+    background-color: #007af4;
+    font-weight: 700;
+    padding: 12px 25px;
+    text-decoration: none;
+    border-radius: 25px;
+    color: #fff;
+    display: inline-block;
+}
```
Comment (andriyParashchuk) [inline themes/beaver/assets/css/use-cases-dynamic.css]: Please add empty line to the end of file

**Review by andriyParashchuk — COMMENTED** [review]
Please make cleanup. 
Please keep  to the one style of indentation between selectors, one space between selector name, use 2 spaces instead 4.

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 3 above)
Comment (andriyParashchuk) [reply themes/beaver/assets/css/use-cases-dynamic.css]: Please be a careful with !iportants. In the case is not needed

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 4 above)
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: @pftg needs only on html files!

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 3 above)
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: ++

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 2 above)
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: ++

**Comment by AucT** [comment]
> Please make cleanup. Please keep to the one style of indentation between selectors, one space between selector name, use 2 spaces instead 4.

++

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 4 above)
Comment (pftg) [reply themes/beaver/assets/css/use-cases-dynamic.css]: @AucT is it a question?

Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: @pftg nope

Comment (andriyParashchuk) [reply themes/beaver/assets/css/use-cases-dynamic.css]: @AucT https://gist.github.com/OleksiyRudenko/d51388345ea55767b7672307fe35adf3

**Review by pftg — CHANGES_REQUESTED** [review]
all tabs are highlighted when it should be only one 
jetthoughts github io – Commit: _use-cases png 2024-11-07 20-31-33

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+  padding-left: 0;
+}
+
+.use-cases .tab-panel .item {
+  background-image: url({{ (resources.Get "img/icons/theme/homepage_tick_circle.svg" | fingerprint).RelPermalink }});
+  background-repeat: no-repeat;
+  background-position: 0 50%; /* Center the bullet vertically */
+  background-size: 24px 24px; /* Size of the bullet */
+  padding-left: 36px; /* Space for the bullet */
+}
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css:162]: nope

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    padding-left: 50px
+  }
+}
+
+@media (max-width: 860px) {
+  .use-cases .tab-panels {
+    padding-left: 0
+  }
+}
+
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: nope

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 7]
```diff
@@ -0,0 +1,146 @@
...
+.use-cases .action-button {
+  background-color: #007af4;
+  font-weight: 700;
+  padding: 12px 25px;
+  text-decoration: none;
+  border-radius: 25px;
+  color: #fff;
+  display: inline-block;
+}
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: nope

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 4 above)
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: @andriyParashchuk ++

File: themes/beaver/assets/js/tabs.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+                }
+            } else if (e.key === 'Enter' || e.key === ' ') {
+                tabs[tabFocus].click();
+            }
+
+            tabs[tabFocus].setAttribute("tabindex", 0);
+            tabs[tabFocus].focus();
+        }
+    });
+});
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/js/tabs.js]: nope

File: themes/beaver/assets/js/tabs.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+        container.querySelectorAll(".tab-panel").forEach(panel => panel.hidden = true);
+        panel.hidden = false;
+        container.querySelectorAll(".js-tab").forEach(tabTitle => tabTitle.setAttribute('aria-selected', 'false'));
+        targetTab.setAttribute('aria-selected', 'true');
+        if (window.innerWidth < 860) {
+            panel.scrollIntoView({behavior: 'smooth', block: 'center'});
+            panel.focus();
+        }
+    }
+});
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/js/tabs.js:15]: nope

File: themes/beaver/assets/js/tabs.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
+document.querySelectorAll('.js-tab').forEach(el => {
+    const container = el.closest('.js-tabs-container');
+    el.onclick = function (e) {
+        const targetTab = e.target;
+        const panel = document.getElementById(targetTab.getAttribute('aria-controls'));
+        container.querySelectorAll(".tab-panel").forEach(panel => panel.hidden = true);
+        panel.hidden = false;
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/js/tabs.js:7]: nope

File: themes/beaver/layouts/page/use-cases.html [hunk 11]
```diff
@@ -1,10 +1,22 @@
...
 {{ end }}
+
+{{ define "footer" }}
+{{ $tabs := slice
+(resources.Get "js/tabs.js")
+| resources.Concat "js/_tabs.js"
+| js.Build (dict "minify" true) | minify | fingerprint | resources.PostProcess
+}}
+<script defer src="{{ $tabs.RelPermalink }}"></script>
+{{ end }}
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/layouts/page/use-cases.html:18]: ++

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 12]
```diff
@@ -0,0 +1,149 @@
...
+.use-cases .tab-title {
+    background-color: #f5f6f8;
+    color: #121212;
+    border: unset !important;
+    border-radius: 14px;
+    padding: 28px 35px;
+    font-size: 20px;
+    font-weight: 700;
+    cursor: pointer;
+}
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: nope

File: themes/beaver/assets/css/use-cases-dynamic.css [hunk 13]
```diff
@@ -0,0 +1,149 @@
...
+
+.use-cases .tab-panel .item  {
+    background-image: url({{ (resources.Get "img/icons/theme/homepage_tick_circle.svg" | fingerprint).RelPermalink }});
+
+background-repeat: no-repeat;
+background-position: 0 50%; /* Center the bullet vertically */
+background-size: 24px 24px; /* Size of the bullet */
+padding-left: 36px; /* Space for the bullet */
+}
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: nope

File: themes/beaver/assets/css/use-cases-dynamic.css (hunk 12 above)
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/css/use-cases-dynamic.css]: ++

File: themes/beaver/assets/js/tabs.js [hunk 14]
```diff
@@ -0,0 +1,12 @@
+document.querySelectorAll('.js-tab').forEach(el => {
+    const container = el.closest('.js-tabs-container');
+    el.onclick = function (e) {
+        const panel = document.getElementById(e.target.getAttribute('aria-controls'));
+        container.querySelectorAll(".tab-panel").forEach(panel => panel.hidden = true);
+        panel.hidden = false;
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/js/tabs.js]: nope

**Comment by AucT** [comment]
> all tabs are highlighted when it should be only one jetthoughts github io – Commit: _use-cases png 2024-11-07 20-31-33

++

File: themes/beaver/assets/js/tabs.js [hunk 15]
```diff
@@ -0,0 +1,46 @@
...
+    }
+});
+
+//for keyboard
+document.querySelectorAll('.tabs').forEach(tabList => {
+    const tabs = tabList.querySelectorAll('.js-tab');
+    let tabFocus = 0;
+    tabList.addEventListener("keydown", (e) => {
+        e.preventDefault();
+        if (e.key === "ArrowDown" || e.key === "ArrowUp" || e.key === 'Enter' || e.key === ' ') {
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/assets/js/tabs.js]: ++

File: themes/beaver/layouts/page/use-cases.html [hunk 16]
```diff
@@ -14,278 +26,11 @@
...
-                                              </div>
-                                           </div>
-                                        </div>
-                                     </div>
-                                  </div>
-                               </div>
-                            </div>
+                         <div class="fl-row-content-wrap">
+                         <div class="fl-row-content fl-row-fixed-width fl-node-content">
+                         {{ partial "page/use-cases.html" (dict "addHeadline" "false") }}
```
[the remark this answers is not in the document]
Comment (AucT) [reply themes/beaver/layouts/page/use-cases.html]: ++

**Review by pftg — CHANGES_REQUESTED** [review]
some ui breaking changes 
Accelerate   Achieve Your Product Vision 2024-11-08 11-20-41

**Comment by AucT** [comment]
> some ui breaking changes Accelerate Achieve Your Product Vision 2024-11-08 11-20-41

added mobile tabs
GH36Hkptuf

**Pull request merged**